src/lamplib/src/genny/tasks/auto_tasks.py

```diff
@@ -449,19 +425,23 @@ def __init__(self, lister: WorkloadLister, reader: YamlReader, workspace_root: s
         self.workspace_root = workspace_root
         self.lister = lister
         self.reader = reader
+        self._all_workloads = None
+
 
     def all_workloads(self) -> List[Workload]:
         all_files = self.lister.all_workload_files()
         modified = self.lister.modified_workload_files()
-        return [
-            Workload(
-                workspace_root=self.workspace_root,
-                file_path=fpath,
-                is_modified=fpath in modified,
-                reader=self.reader,
-            )
-            for fpath in all_files
-        ]
+        if self._all_workloads is None:
```

Comment:
Should you move the first two lines inside the if? You don't need to do that work if `self._all_workloads` exist.

Reply:
Yes definitely

Reply:
Fixed